src/runtime_src/xdp/profile/database/static_info_database.cpp

```diff
@@ -2684,11 +2700,12 @@ namespace xdp {
     if (!xclbin)
       return;
 
-    if (!metadataReader)
+    if (!getAIEmetadataReader(deviceId))
        return;
 
     try {
-      xclbin->aie.clockRateAIEMHz = metadataReader->getAIEClockFreqMHz();
+      std::lock_guard<std::mutex> lock(aieMetadataReaderLock) ;
```

Comment:
We are already fetching metadataReader for "deviceId" at Line 2703. Please reuse that and remove the aiemetadatareader lock here.